src/wsrep_provider_v26.cpp

```diff
@@ -394,12 +395,34 @@ namespace
         }
     }
     
-    int encrypt_cb(wsrep_enc_ctx_t*      /*ctx*/,
-                   const wsrep_buf_t*    /*input*/,
-                   void*                 /*output*/,
-                   wsrep_enc_direction_t /*direction*/,
-                   bool                  /*final*/)
+    wsrep_cb_status_t encrypt_cb(void*                 app_ctx,
+                                 wsrep_enc_ctx_t*      enc_ctx,
+                                 const wsrep_buf_t*    input,
+                                 void*                 output,
+                                 wsrep_enc_direction_t direction,
+                                 bool                  final)
     {
+        assert(app_ctx);
+        wsrep::server_state& server_state(
+            *reinterpret_cast<wsrep::server_state*>(app_ctx));
+
+        wsrep::const_buffer key(enc_ctx->key->ptr, enc_ctx->key->len);
+        wsrep::const_buffer in(input->ptr, input->len);
+
+        try
+        {   
+            server_state.server_service().do_crypt(&enc_ctx->ctx,
```

Comment:
I'm not sure involving server_service interface here is a good idea, but it needs to be discussed. Compare it to `sst_request_cb` above.

Reply:
If the call does not require any access or modification to server_state internal state, a direct call to server_service method is good.

Reply:
Ok, then it is good. My main concern here was introduction of dependence on the `server_service` interface, but if it is conceptually correct, which it seems to be, then it is good.